R/stat_cross.R

```diff
@@ -301,3 +301,72 @@ ggally_tableDiag <- function(data, mapping, ..., geom_tile_args = NULL) {
   mapping$y <- mapping$x
   ggally_table(data = data, mapping = mapping, ..., geom_tile_args = geom_tile_args)
 }
+
+#' Display a cross-tabulated table
+#'
+#' \code{ggally_table2} is a variation of \code{ggally_table} with few modifications: (i) table cells are drawn; (ii) x and y axis are not expanded (and therefore are not aligned with other \code{ggally_*} plots); (iii) content and fill of cells can be easily controlled with dedicated arguments.
+#' @param data data set using
+#' @param mapping aesthetics being used
+#' @param cells Which statistic should be displayed in table cells?
+#' @param fill Which statistic should be used for filling table cells?
+#' @param ... other arguments passed to \code{\link[ggplot2]{geom_text}(...)}
+#' @param geom_tile_args other arguments passed to \code{\link[ggplot2]{geom_tile}(...)}
+#' @importFrom scales percent
+#' @importFrom scales number
+#' @export
+#' @examples
+#' # ggally_table2() examples
+#' ggally_table(tips, mapping = aes(x = day, y = sex))
+#' ggally_table2(tips, mapping = aes(x = day, y = sex))
+#'
+#' # display column proportions
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), cells = "col.prop")
+#'
+#' # display row proportions
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), cells = "row.prop")
+#'
+#' # change size of text
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), size = 8)
+#'
+#' # fill cells with standardized residuals
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), fill = "stdres")
+#'
+#' \dontrun{
+#' # change scale for fill
+#' ggally_table2(tips, mapping = aes(x = day, y = sex), fill = "stdres") +
+#'   scale_fill_steps2(breaks = c(-2, 0, 2), show.limits = TRUE)
+#' }
+ggally_table2 <- function(
```

Comment:
May we call this `ggally_crosstable` (or `ggally_cross`)? ... Something without the `2`.  When I see the `2`, I interpret that I should not use the original.

Reply:
names changed for ggally_crosstable